typing/env.mli

```diff
@@ -360,6 +360,14 @@ val bound_cltype: string -> t -> bool
 
 val make_copy_of_types: t -> (t -> t)
 
+(* Resolution of globals *)
+
+(* [global_of_instance_compilation_unit cu] checks that a compilation unit is a
+   complete instantiation - that is, that all of its parameters are filled by
+   arguments, and all of those arguments' parameters are filled, and so on -
+   and converts it into a global. *)
+val global_of_instance_compilation_unit : Compilation_unit.t -> Global_module.t
```

Comment:
I think this change is related to code-generation and should stay in this PR. The rest of changes in this file should go to the previous PR.